### PR TITLE
xbps-src: add make_check=ci-skip

### PR DESCRIPTION
**Pull request by Chocimier**
For packages with tests failing as root or similar.

`test $XBPS_CHROOT_CMD = ethereal` could be used in practice instead of XBPS_BUILD_ENVIRONMENT, but I like that way more for possibility of configuration local env as ci, and not conflating CI with chroot style.

Some packages are not tested locally yet.

### CODE REVIEW

File: Manual.md [hunk 1]
```diff
@@ -606,10 +606,15 @@ patches to the package sources during `do_patch()`. Patches are stored in
...
 
-- `make_check` Sets the cases in which the `check` phase is run. Can be `yes` (the default) to run if
-`XBPS_CHECK_PKGS` is set, `extended` to run if `XBPS_CHECK_PKGS` is `full` and `no` to never run.
+- `make_check` Sets the cases in which the `check` phase is run.
 This option should usually be accompanied by a comment explaining why it was set, especially when
 set to `no`.
+Allowed values:
+  - `yes` (the default) to run if `XBPS_CHECK_PKGS` is set.
+  - `extended` to run if `XBPS_CHECK_PKGS` is `full`.
+  - `ci-skip` to run locally if `XBPS_CHECK_PKGS` is set, but not as part of pull request checks.
```
Comment (ericonr) [inline Manual.md:615]: And if there are long and short tests, one can still differentiate manually by checking the value of `XBPS_CHECK_PKGS` in `do_check`, right?

Might make sense to note that such a mechanism is available in general, maybe.

But I overall like that tests for `ci-skip` would run with `./xbps-src -Q` :)

File: common/xbps-src/libexec/xbps-src-docheck.sh [hunk 2]
```diff
@@ -43,6 +43,12 @@ if [ "$make_check" = extended -a "$XBPS_CHECK_PKGS" != full ]; then
     exit 0
 fi
 
+if [ "$make_check" = ci-skip ] && [ "$XBPS_BUILD_ENVIRONMENT" = void-packages-ci ]; then
+    msg_normal \
+        "${pkgname}-${version}_${revision}: skipping check (make_check=ci-skip and XBPS_BUILD_ENVIRONMENT=void-packages-ci) ...\n"
+    exit 0
+fi
+
```
Comment (ericonr) [inline common/xbps-src/libexec/xbps-src-docheck.sh:51]: I think these should jump at reviewers and submitters; I would use a `msg_warn` and add "make sure to run tests locally" (or something to that effect, my phrasing isn't great) to the message.

File: etc/defaults.conf [hunk 3]
```diff
@@ -136,6 +136,12 @@ XBPS_SUCMD="sudo /bin/sh -c"
 #
 #XBPS_USE_BUILD_MTIME=yes
 
+# [OPTIONAL]
+# Enable CI specific mode of operation. Currently this disables do_check for some
```
Comment (ericonr) [inline etc/defaults.conf:140]: I think spelling "continuous integration (CI)" out loud is valid here.

**Review by ericonr — COMMENTED** [review]
Overall I like the change, but this makes remembering to run tests locally important, so I think `msg_warn` is a valid usage here.

If possible at some point in the future, I'd even consider adding a checkbox to the pull request template to ask if the test suite has been run (short term) and have a bot check for it with the `make_check=ci-skip` packages (wishlist).

Maybe some of the comments on top of `make_check=ci-skip` can be removed now, since it's fairly self explanatory?

**Comment by ericonr** [comment]
Ah, I'd also consider doing a single commit for all the packages :)

The change is very homogeneous and makes it clear that they are a group.

File: Manual.md (hunk 1 above)
Comment (Chocimier) [reply Manual.md:614]: Yes, right. XBPS_CHECK_PKGS was described above, reworded a bit.

**Review by Chocimier — COMMENTED** [review]
Changed. Modifying multiple templates is against rules, isn't it?

**Comment by ericonr** [comment]
> Modifying multiple templates is against rules, isn't it?

Depends, if it's a change that's changing the same thing in multiple packages *and* doesn't change the binary package and doesn't revbump anything, it makes sense to group together. If we were rebuilding any of them I'd do separate commits.

**Pull request merged**